tools/odbc/test/tests/col_attribute/char_query.cpp

```diff
@@ -0,0 +1,49 @@
+#include "utils.h"
+
+using namespace odbc_col_attribute_test;
+
+TEST_CASE("Test SQLColAttribute for a query that returns a char", "[odbc]") {
+	SQLHANDLE env;
+	SQLHANDLE dbc;
+
+	HSTMT hstmt = SQL_NULL_HSTMT;
+
+	// Connect to the database using SQLConnect
+	CONNECT_TO_DATABASE(env, dbc);
+
+	// Allocate a statement handle
+	EXECUTE_AND_CHECK("SQLAllocHandle (HSTMT)", SQLAllocHandle, SQL_HANDLE_STMT, dbc, &hstmt);
+
+	// run a simple query with chars to get a result set
+	EXECUTE_AND_CHECK("SQLExecDirect", SQLExecDirect, hstmt, ConvertToSQLCHAR("SELECT 'a' AS a, 'b' AS b"), SQL_NTS);
+	std::map<SQLLEN, ExpectedResult *> expected_chars;
+	expected_chars[SQL_DESC_CASE_SENSITIVE] = new ExpectedResult(SQL_TRUE);
```

Comment:
Just curious as to why we use a pointer in the map and `new` here. Can we not insert `ExpectedResult` directly?

Reply:
Because if `ExpectedResult` is not a pointer, in TestAllFields the program tries to initialize it instead of reading the value stored, so I couldn't get around using `new`.

Reply:
Let's check if the element exists in the map and only then access it with the iterator. Would that work?
```cpp
 auto it = my_map.find('key');
 if (it != my_map.end()) { // get *it.second and go things
  ```